src/net_processing.cpp

```diff
@@ -2411,6 +2411,12 @@ bool ProcessMessage(CNode& pfrom, const std::string& msg_type, CDataStream& vRec
                       pfrom.nVersion.load(), pfrom.nStartingHeight,
                       pfrom.GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom.addr.ToString()) : ""),
                       pfrom.m_tx_relay == nullptr ? "block-relay" : "full-relay");
+            if (!pfrom.m_tx_relay) {
+                std::vector<CAddress> anchors_to_save = connman->GetBlockRelayNodeAddresses();
+                const std::vector<CAddress> anchors_not_tried = connman->GetAnchors();
```

Comment:
I'm a bit confused of what's going on here. I guess this is supposed to address the issue of reconnecting to the same anchor on many restarts, but it would really use some extra comments.

Reply:
I've tried to avoid any confusion by meaningful naming variables, but failed :)
The node dumps:
- anchors that are connected right now
- anchors that were known at startup but have not tried to connect to yet

The latter set quickly gets empty after startup.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/17428#issuecomment-653533271).